src/python/pants/engine/legacy/options_parsing.py

```diff
@@ -0,0 +1,48 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.engine.rules import RootRule, rule
+from pants.engine.selectors import Select
+from pants.init.options_initializer import OptionsInitializer
+from pants.option.options_bootstrapper import OptionsBootstrapper
+from pants.util.objects import datatype
+
+
+class OptionsParseRequest(datatype(['args', 'env'])):
+  """Represents a request for Options computation."""
+
+  @classmethod
+  def create(cls, args, env):
+    assert isinstance(args, (list, tuple))
+    return cls(
+      tuple(args),
+      tuple(env.items() if isinstance(env, dict) else env)
+    )
+
+
+class Options(datatype(['options', 'build_config'])):
+  """Represents the result of an Options computation."""
+
+
+@rule(Options, [Select(OptionsParseRequest)])
```

Comment:
So, given that `OptionsParseRequest` doesn't contain the contents of `pants.ini` files, somewhere inside of `options_initializer.setup` we must be touching the filesystem to read those.

In order to safely memoize this `@rule` (ie, use it in `pantsd`), we'd need to move that filesystem access to usage of `PathGlob -> FileContents`.

Reply:
will do!